k8s.jsonnet

```diff
@@ -158,23 +187,23 @@
   // (they may duplicate work, but not cause errors by stepping on each others' toes).
   items: [
     // The downloader watches the twitch stream and writes the HLS segments to disk
-    $.deployment("downloader", args=$.config.channels + [
+    if $.config.enabled["downloader"] then $.deployment("downloader", args=$.config.channels + [
```

Comment:
Note that a jsonnet expression like `if COND then EXPR` without an else will resolve to `null` when COND is false.
That means your `items` list will end up with `null` items in it.

Two ways to work around this:
* do `if COND then [ITEM] else []` to produce an array of either 1-item or 0-item arrays, then flatten
* filter out the nulls later (eg. `[item for item in list if item != null]`)

IMO the latter is simpler and easier to read.